tests/E2E Tests/TokenAcquirerTests/TokenAcquirer.cs

```diff
@@ -39,6 +39,7 @@ public TokenAcquirer()
         [Fact]
         public void TokenAcquirerFactoryDoesNotUseAspNetCoreHost()
         {
+            TokenAcquirerFactory.ResetDefaultInstance();
```

Comment:
Doesn't xunit provide a way to run some code before each test? Seems like you already have a ctor with this logic, but the ctor only runs once for all tests in this class.

https://stackoverflow.com/questions/46926852/xunit-constructor-runs-before-each-test

Reply:
It does, but most of these tests aren't run in DevOps build so I didn't want to include them in the scope of this PR for the sake of the two that are.